scripts/edgepy.py

```diff
@@ -37,39 +59,177 @@ def __init__(self, args):
+                key_name = args.mongo_key_name
+                key_value = args.mongo_key_value
+            else:
+                raise ValueError("Insufficient parameters for use of Mongodb")
+
             mongo_importer = ImportFromMongodb(
                 host=config.get("Mongo", "host"),
                 port=config.get("Mongo", "port"),
-                mongo_key_name=args.mongo_key_name,
-                mongo_key_value=args.mongo_key_value,
+                mongo_key_name=key_name,
+                mongo_key_value=key_value,
                 gene_list_file=args.gene_list,
             )
+
             sample_list, data_set, gene_list, sample_category = (
                 mongo_importer.get_data_from_mongo()
             )
+
+            if key_name == 'sample_name':
+                # Override sample categories if sample name is the source of the categories.
+                sample_category_list = [
+                    "group1" if sample_name in args.group1_sample_names else "group2"
+                    for sample_name in sample_list
+                ]
+                sample_category_dict = None
+            else:
+                # TODO: read from file
+                sample_category_dict = args.groups_json
+                sample_category_list = None
+
             self.dge_list = DGEList.create_DGEList(
-                sample_list, data_set, gene_list, sample_category
+                sample_list,
+                data_set,
+                gene_list,
+                sample_category_list=sample_category_list,
+                sample_category_dict=sample_category_dict,
             )
 
-            self.dge_list.write_npz_file("./edgePy/data/example_data.cpe")
+            self.ensg_to_symbol = mongo_importer.mongo_reader.find_as_dict(
+                'ensembl_90_37', "symbol_by_ensg", query={}
+            )
 
         else:
             self.dge_list = DGEList.create_DGEList_data_file(
                 data_file=args.counts_file, group_file=args.groups_file
             )
 
-    def run(self):
-        pass
+        self.output = args.output if args.output else None
+        self.p_value_cutoff = args.cutoff
+        self.minimum_cpm = args.minimum_cpm
+
+    def fake_groups(self):
+        """
+        This is used to create fake groups, if you didn't give any at the time the DGEList object was
+         created... this is a bug that needs to be fixed.
+        :return:
+        """
+        self.dge_list.group = [
+            'fibroblast' if 'fibroblast' in g else 'cardiomyocyte' for g in self.dge_list.group
+        ]
+
+    def run_ks(self):
+        """
+        First pass implementation of a Kolmogorov-Smirnov test for different groups, using the Scipy KS test two-tailed
+        implementation.
+
+        Args:
+             None.
+        """
+
+        print(self.dge_list.groups_list)
+
+        gene_details, gene_likelyhood1, group_types = self.ks_2_samples()
```

Comment:
sp? likelihood.

Reply:
This misspelling appears elsewhere too FYI

Reply:
fixed.